### PR TITLE
Support PL 1.8

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes # (issue)

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [ ] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [ ] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,2 @@
 torch>=1.9.*
-torchmetrics>=0.4.1
-pytorch-lightning>=1.6.0
```
Comment (rohitgr7) [inline requirements.txt:3]: TODO: add requirements back

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1 @@
-torch>=1.9.*
-torchmetrics>=0.4.1
-pytorch-lightning>=1.6.0
-lightning-utilities==0.3.0
-packaging
+pytorch-lightning>=1.8.0
```
Comment (Borda) [inline requirements.txt:1]:
```suggestion
pytorch-lightning>=1.8.0 --pre
```

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1 @@
-torch>=1.9.*
-torchmetrics>=0.4.1
-pytorch-lightning>=1.6.0
-lightning-utilities==0.3.0
-packaging
+pytorch-lightning>=1.8.0 --pre
```
Comment (Borda) [inline requirements.txt:1]:
```suggestion
pytorch-lightning==1.8.0rc1  # fixme: use normal release
```

**Comment by Borda** [comment]
@rohitgr7, how is it going here? :)

**Comment by rohitgr7** [comment]
tests are good now. Only 2 ci blockers:
1. Lightning 1.8 release. Since we shouldn't keep rc in requirements.
2. https://github.com/Lightning-AI/lightning/issues/15368 which is causing a ci failure.

**Comment by akihironitta** [comment]
There're two things that currently blocking Bolts' GPU CI:
- https://github.com/Lightning-AI/lightning-bolts/issues/922 (which I'm working on)
- https://github.com/Lightning-AI/lightning-bolts/pull/912

cc @rohitgr7

**Comment by rohitgr7** [comment]
@akihironitta not sure about the second one because it's failing locally too.. let's see. I have re-ran the tests there

**Review by awaelchli — APPROVED** [review]

**Pull request merged**